ci failed

### Which tests are failing?

https://github.com/spidernet-io/spiderpool/actions/runs/7060534656/job/19220444556

### Since when has it been failing?

https://github.com/spidernet-io/spiderpool/actions/runs/7060534656/job/19220444556

### Job's Link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

```text
/etc/bash.bashrc: line 7: PS1: unbound variable
install-default-cni.sh : E2E_CLUSTER_NAME spiderpool1201132400 
install-default-cni.sh : INSTALL_CALICO true 
install-default-cni.sh : INSTALL_CILIUM false 
install-default-cni.sh : CLUSTER_PATH /home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool1201132400 
failed get calico version
make[2]: *** [Makefile:21: kind-init] Error 1
make[2]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make[1]: *** [Makefile:298: e2e_init] Error 2
make[1]: Leaving directory '/home/runner/work/spiderpool/spiderpool'
make: *** [Makefile:319: e2e_init_calico] Error 2
```

--- a/test/Makefile.defs
+++ b/test/Makefile.defs
@@ -130,6 +130,7 @@ E2E_MULTUS_IMAGE_NAME ?= $(E2E_MULTUS_IMAGE_REGISTER)/k8snetworkplumbingwg/multu
 
 #================= calico
 CALICO_VERSION ?=
+DEFAULT_CALICO_VERSION ?= v3.26.4
 
 ifeq ($(E2E_CHINA_IMAGE_REGISTRY),true)
   CALICO_IMAGE_REPO ?= docker.m.daocloud.io

--- a/test/scripts/install-default-cni.sh
+++ b/test/scripts/install-default-cni.sh
@@ -56,9 +56,9 @@ CILIUM_CLUSTER_POD_SUBNET_V6=${CILIUM_CLUSTER_POD_SUBNET_V6:-"fd00:10:244::/112"
 function install_calico() {
     cp ${PROJECT_ROOT_PATH}/test/yamls/calico.yaml $CLUSTER_PATH/calico.yaml
     if [ -z "${CALICO_VERSION}" ]; then
-      [ -n "${HTTP_PROXY}" ] && CALICO_VERSION=$(curl --retry 3 -x "${HTTP_PROXY}" -s https://api.github.com/repos/projectcalico/calico/releases/latest | jq -r '.tag_name')
-      [ -z "${HTTP_PROXY}" ] && CALICO_VERSION=$(curl --retry 3 -s https://api.github.com/repos/projectcalico/calico/releases/latest | jq -r '.tag_name')
-      [ "${CALICO_VERSION}" == "null" ] && { echo "failed get calico version"; exit 1; }
+      [ -n "${HTTP_PROXY}" ] && { CALICO_VERSION_INFO=$(curl --retry 3 --retry-delay 5 -x "${HTTP_PROXY}" -s https://api.github.com/repos/projectcalico/calico/releases/latest); echo ${CALICO_VERSION_INFO}; CALICO_VERSION=$(echo ${CALICO_VERSION_INFO} | jq -r '.tag_name'); }
+      [ -z "${HTTP_PROXY}" ] && { CALICO_VERSION_INFO=$(curl --retry 3 --retry-delay 5 -s https://api.github.com/repos/projectcalico/calico/releases/latest ); echo ${CALICO_VERSION_INFO}; CALICO_VERSION=$(echo ${CALICO_VERSION_INFO} | jq -r '.tag_name'); }
+      [ "${CALICO_VERSION}" == "null" ] && { echo "failed to get the calico version, will try to use default version."; CALICO_VERSION=${DEFAULT_CALICO_VERSION}; }
     else
       CALICO_VERSION=${CALICO_VERSION}
     fi

